conan/internal/graph/graph.py

```diff
@@ -342,7 +342,9 @@ def items(self):
         return self._overrides.items()
 
     def serialize(self):
-        return {k.repr_notime(): [e.repr_notime() if e else None for e in v]
+        return {k.repr_notime():
+                    sorted([e.repr_notime() if e else None for e in v],
```

Comment:
I see the ``Overrides.deserialize()`` contains a ``set([RecipeReference.loads(e) if e else None for e in v])``, so it seems we can apply here ordering without much risk